ledger/src/blockstore_db.rs

```diff
@@ -441,17 +453,17 @@ impl Rocks {
     }
 
     fn get_cf(&self, cf: &ColumnFamily, key: &[u8]) -> Result<Option<Vec<u8>>> {
-        let opt = self.db.get_cf(cf, key)?;
+        let opt = rocks_try!(self.db.get_cf(cf, key));
```

Comment:
why `get_cf`/`put_cf`/`delete_cf`/`delete_file_in_range_cf` are only chosen to look for corruption errors?

are we certain this is the only source of the error?

Reply:
also, i think this is a bit maintenance annoyance

instead, i'd propose remove this auto-derived `From` trait impl and write our own one.in that way, type safety may bless us. :) In that custom impl, the fn would take `(err, atomicbool)` and set `true` to the bool if `ErrorKind::Corruption` is detected.

https://github.com/solana-labs/solana/blob/1eba91af2928848dc9a861d1af351434d0752d0c/ledger/src/blockstore_db.rs#L104

Reply:
> why get_cf/put_cf/delete_cf/delete_file_in_range_cf/write are only chosen to look for corruption errors?
> are we certain this is the only source of the error?
These are just starters :p.  Ideally, it should apply to all rocksdb calls.  This PR only takes care of those rocksdb calls inside `struct Rocks`.

Reply:
There's another one that we cannot catch is the corruption reported via rocksdb's background compaction job.

Reply:
> instead, i'd propose remove this auto-derived From trait impl and write our own one.in that way, type safety may bless us. :) In that custom impl, the fn would take (err, atomicbool) and set true to the bool if ErrorKind::Corruption is detected.

This sounds like a good idea.  Let me think more about it as we want it to be general enough to cover all types of severe/irrecoverable errors.

Reply:
> There's another one that we cannot catch is the corruption reported via rocksdb's background compaction job.

oh, that's good point. btw, can we be notified these errors in some way like registering error listeners to rocks? if it's easy, let's do this. :)

Reply:
> oh, that's good point. btw, can we be notified these errors in some way like registering error listeners to rocks? if it's easy, let's do this. :)

This could be done via the `EventListener::OnBackgroundError()` callback, but the EventListener API isn't available in RocksDB's C API (and thus not available in the rust-rocksdb API).